profile/httprouter/handler.go

```diff
@@ -0,0 +1,53 @@
+// SPDX-FileCopyrightText: 2022 Open Networking Foundation <info@opennetworking.org>
+// Copyright 2019 free5GC.org
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package httprouter
+
+import (
+	"net/http"
+
+	"github.com/free5gc/openapi"
+	"github.com/free5gc/openapi/models"
+	"github.com/gin-gonic/gin"
+	"github.com/omec-project/gnbsim/logger"
+	profile "github.com/omec-project/gnbsim/profile"
+	profCtx "github.com/omec-project/gnbsim/profile/context"
+)
+
+func HTTPExecuteProfile(c *gin.Context) {
+
+	logger.HttpLog.Infoln("EcecuteProfile API called")
+	var prof profCtx.Profile
+
+	requestBody, err := c.GetRawData()
+	if err != nil {
+		logger.HttpLog.Errorf("Get Request Body error: %+v", err)
+		problemDetail := models.ProblemDetails{
+			Title:  "System failure",
+			Status: http.StatusInternalServerError,
+			Detail: err.Error(),
+			Cause:  "SYSTEM_FAILURE",
+		}
+		c.JSON(http.StatusInternalServerError, problemDetail)
+		return
+	}
+
+	err = openapi.Deserialize(&prof, requestBody, "application/json")
+	if err != nil {
+		problemDetail := "[Request Body] " + err.Error()
+		rsp := models.ProblemDetails{
+			Title:  "Malformed request syntax",
+			Status: http.StatusBadRequest,
+			Detail: problemDetail,
+		}
+		logger.HttpLog.Errorln(problemDetail)
+		c.JSON(http.StatusBadRequest, rsp)
+		return
+	}
+	logger.HttpLog.Debugf("%#v", prof)
+
+	prof.Init()
+	go profile.ExecuteProfile(&prof, profCtx.SummaryChan)
```

Comment:
what is http response code ?

Reply:
If execution reaches line 52, then the handler has been successfully executed, and by default, it returns 200
(I have updated the README to add '-i' option in the curl command which prints the received response code)

However before line 52, if the error occurs, then it is been notified either with response code 400 or 500 (depending on the type of error)